clowder-framework/clowder
# [WIP] Feature: Extractor Catalog

## Description
![Extractor Catalog Design - Catalog View](https://user-images.githubusercontent.com/1413653/91781476-03e7ea00-ebc0-11ea-9cb5-e3de188581f6.png)

<img width="1429" alt="Screen Shot 2020-08-31 at 7 54 57 PM" src="https://user-images.githubusercontent.com/1413653/91782870-02202580-ebc4-11ea-9b4b-e5f0edebe9a0.png">

### Problem
We currently have a very primitive table view for our Extractor Catalog. We plan on evolving it into a more capable view, but there may be merge conflicts if too many things happen here at once.

### Approach
The purpose of this PR is to (hopefully) minimize conflict down the road by creating a basic layout and stubbing out any missing logic.

This task is blocking near-future development towards the Extractor Catalog.

### How to Test
Prerequisites: 2 users (1 admin + 1 non-admin), at least one extractor registered

1. Checkout and run this branch locally
2. Sign-up a non-admin user (if you haven't already) and login to Clowder as non-admin
3. Enable the RabbitMQ plugin and register an extractor, if you haven't already
4. From the navbar, choose "Explore > Extractors"
    * You should be brought to a read-only version of the Extractor Catalog
    * You should see each extractor's Name, Authors, Version, Maturity (stubbed), My/All Job Counts (stubbed), and Ratings (stubbed)
    * You should see that a dropdown menu is offered for each extractor
    * You should see a Labels and Filters section
    * You should be able to see (but not modify) which extractors are currently enabled on Clowder
    * You should **not** see a "Manage" button at the top-right of the Labels section
    * You should **not** see any buttons beneath the list of extractors
5. Expand the dropdown menu beside an extractor
    * You should see a dropdown menu appear
    * You should see the dropdown offers a working option to "View Details"
    * You should see the dropdown offers (stubbed) options to "View History", "View Logs", "View Metrics", and "Rate & Comment"
    * You should **not** see (stubbed) options to "Move to Development", "Move to Staging", or "Move to Production" (these are admin-only)
6. Logout and log back into Clowder as an admin
7. From the navbar, choose "Explore > Extractors"
    * You should be brought to the writable version of the enable/disable extractors list which preserves existing admin functionality
    * You should be able to check/uncheck which extractors are currently enabled on Clowder
    * You should see each extractor's Name, Authors, Version, Maturity (stubbed), My/All Job Counts (stubbed), and Ratings (stubbed)
    * You should see that a dropdown menu is offered for each extractor
    * You should see a Labels and Filters section, with a "Manage" button at the top-right of the Labels section (admin-only)
    * You should see Save/Cancel buttons beneath the list of extractors
8. Expand the dropdown menu beside an extractor
    * You should see a dropdown menu appear
    * You should see the dropdown offers a working option to "View Details"
    * You should see the dropdown offers (stubbed) options to "View History", "View Logs", "View Metrics", and "Rate & Comment"
    * You should see (stubbed) options to "Move to Development", "Move to Staging", or "Move to Production" (these are admin-only)

## Review Time Estimate


## Types of changes


## Checklist:
